www/js/lib/zimArchive.js

```diff
@@ -142,31 +142,71 @@ define(['zimfile', 'zimDirEntry', 'util', 'utf8'],
      */
 
     /**
-     * Look for DirEntries with title starting with the given prefix.
+     * Look for DirEntries with title starting with the prefix of the current search object.
      * For now, ZIM titles are case sensitive.
      * So, as workaround, we try several variants of the prefix to find more results.
      * This should be enhanced when the ZIM format will be modified to store normalized titles
      * See https://phabricator.wikimedia.org/T108536
      * 
-     * @param {String} prefix
-     * @param {Integer} resultSize
-     * @param {callbackDirEntryList} callback
+     * @param {Object} search The current state.search object
+     * @param {Integer} resultSize The number of dirEntries to find
+     * @param {callbackDirEntryList} callback The function to call with the result
+     * @param {Boolean} noInterim A flag to prevent callback until all results are ready (used in testing) 
      */
-    ZIMArchive.prototype.findDirEntriesWithPrefix = function(prefix, resultSize, callback) {
+    ZIMArchive.prototype.findDirEntriesWithPrefix = function (search, resultSize, callback, noInterim) {
```

Comment:
As explained above, I think we should pass both the originalPrefix (that never changes) AND the search object, to be able to detect a search replaced by another one.

Reply:
@mossroy This has been done by declaring a `localPrefix` as a constant, which obviously is not changed at any point in the code, and which is then used as the comparator.